### DISCUSSION TITLE
Does powershell expose any ValidateParameterSets functions which can be used in C#?

### DISCUSSION THREAD

**Discussion by anmolrana96** [PowerShell/PowerShell#19441] [Q&A]
I am invoking powershell scripts from c#. But before actual invoke I want to validate parameters user has provided. I am doing that validation by getting all parameter sets but I was wondering that there can be mismatch in behaviour of powershell validation and custom validation that I implement. 
Is there any way to use powershell validation in C# to maintain same behaviour.

**Comment by kasini3000** [comment]
Currently,
The powershell lacks validation of the type 'timespan'.

"Parameter names do not support numeric prefixes,",
https://github.com/PowerShell/PowerShell/issues/17521

You know C #, are you willing to help Powershell pr? 

----

https://learn.microsoft.com/en-us/dotnet/api/system.management.automation?view=powershellsdk-7.0.0

ValidateArgumentsAttribute	
Serves as the base class for Validate attributes that validate parameter arguments.

ValidateCountAttribute	
Validates that the parameter argument count is in the specified range.

ValidateDriveAttribute	
Validates that the path has an approved root drive.

ValidateEnumeratedArgumentsAttribute	
A variant of ValidateArgumentsAttribute which unrolls enumeration values and validates each element individually.

ValidateLengthAttribute	
Validates that the length of each parameter argument's Length falls in the range specified by MinLength and MaxLength.

ValidateNotNullAttribute	
Validates that the parameters's argument is not null.

ValidateNotNullOrEmptyAttribute	
Validates that the parameters's argument is not null, is not an empty string, and is not an empty collection.

ValidatePatternAttribute	
Validates that each parameter argument matches the RegexPattern.

ValidateRangeAttribute	
Validates that each parameter argument falls in the range specified by MinRange and MaxRange.

ValidateScriptAttribute	
Class for validating against a script block.

ValidateSetAttribute	
Validates that each parameter argument is present in a specified set.

ValidateTrustedDataAttribute	
Validates that each parameter argument is Trusted data.

ValidateUserDriveAttribute	
Validates that the path parameter is a User drive.

ValidationMetadataException	
Defines the exception thrown for all Validate attributes.

**Reply by anmolrana96** [reply]
My question is about are there any public methods exposed by powershell sdk to do these validations without invoking commands?
